### PR TITLE
Added delve 'call' support

### LINKED ISSUE #100: debug: support function calls via delve 'call'
Following up from https://github.com/microsoft/vscode-go/issues/2655 as it was the most upvoted feature there and is still badly needed.

# Description
Delve has added the **call** command to be able to make function calls while in debug.

Trying to execute any function in VSCode Debug Console results in:
`Unable to eval expression: "function calls not allowed without using 'call'"`

There was a patch attached in the linked issue that added basic support for this feature. I can add a PR rebased on the latest version of the code to see if it's possible to continue from there.

### PR DESCRIPTION
**Pull request by ekulabuhov**
Fixes #100

### CODE REVIEW

**Comment by hyangah** [comment]
@ekulabuhov - Thank you for the PR! I am so excited about this feature.

Can you please sign the CLA so we can start the review process?
Here is the instruction for contribution (how to write the PR/commit message, etc). https://github.com/golang/vscode-go/blob/v0.15.0-rc.3/docs/contributing.md

**Comment by ekulabuhov** [comment]
@googlebot I signed it!

File: src/debugAdapter/goDebug.ts [hunk 1]
```diff
@@ -1596,30 +1597,42 @@ export class GoDebugSession extends LoggingDebugSession {
 
 	protected evaluateRequest(response: DebugProtocol.EvaluateResponse, args: DebugProtocol.EvaluateArguments): void {
 		log('EvaluateRequest');
-		this.evaluateRequestImpl(args).then(
-			(out) => {
-				const variable = this.delve.isApiV1 ? <DebugVariable>out : (<EvalOut>out).Variable;
-				// #2326: Set the fully qualified name for variable mapping
-				variable.fullyQualifiedName = variable.name;
-				response.body = this.convertDebugVariableToProtocolVariable(variable);
+		const re = new RegExp(/\w+(?=\(.*\))/, 'g');
```
Comment (eliben) [inline src/debugAdapter/goDebug.ts:1600]: Please add a detailed comment here (or above the `evaluateRequest` method) explaining this contract: what exactly we're looking for in the eval request to interpret as a 'call', and how it's treated. A reader writing a test for this should be able to read the comment and understand how to craft test DAP messages that would trigger this behavior, what its limitations and corner cases are, etc.

**Comment by mojtabacazi** [comment]
@eliben @ekulabuhov Can we move this forward?

**Comment by hyangah** [comment]
@ekulabuhov I created a cl based on your change.
https://go-review.googlesource.com/c/vscode-go/+/249377  
I wanted to add you to the cl for review but gerrit doesn't seem to be able to find your account.

FYI Sorry that it would've been ideal if I could push the commit to this branch and the gerritbot could handle this without me creating a separate cl. I filed a feature request here for gerritbot improvement. https://github.com/golang/go/issues/40906

**Comment by hyangah** [comment]
I just submitted https://go-review.googlesource.com/c/vscode-go/+/249377 that started based on @ekulabuhov's PR. 
The feature will be available in the next release. Thanks!
